File: KuantumKaosYonetimi/Javascript/KuantumKaosYonetimi/KuantumCokusuException.js
// KuantumCokusuException.js

/**
 * D. Özel Hata Yönetimi (Custom Exception)
 * Stabilite %0'ın altına düştüğünde fırlatılır.
 */
class KuantumCokusuException extends Error {
    constructor(nesneId) {
        super(`Kuantum Çöküşü! Patlayan Nesne ID: ${nesneId}`);
        this.name = "KuantumCokusuException"; // Hata tipini belirtir
        this.nesneId = nesneId;
    }
}

module.exports = KuantumCokusuException;

File: KuantumKaosYonetimi/Python/KuantumKaosYonetimi/kuantum_cokusu_exception.py
# kuantum_cokusu_exception.py

# D. Özel Hata Yönetimi
class KuantumCokusuException(Exception):
    """
    Stabilite %0'ın altına düştüğünde fırlatılacak özel hata sınıfı.
    """
    def __init__(self, nesne_id):
        self.nesne_id = nesne_id
        super().__init__(f"Kuantum Çöküşü! Patlayan Nesne ID: {nesne_id}")